lib/Doctrine/ORM/EntityManager.php

```diff
@@ -542,35 +542,15 @@ public function detach($entity): void
     }
 
     /**
-     * Merges the state of a detached entity into the persistence context
-     * of this EntityManager and returns the managed copy of the entity.
-     * The entity passed to merge will not become associated/managed with this EntityManager.
+     * Not supported.
      *
-     * @deprecated 2.7 This method is being removed from the ORM and won't have any replacement
+     * @param object $object
      *
-     * @param object $entity The detached entity to merge into the persistence context.
-     *
-     * @return object The managed copy of the entity.
-     *
-     * @throws ORMInvalidArgumentException
-     * @throws ORMException
+     * @psalm-return never
      */
-    public function merge($entity): object
+    public function merge($object): object
```

Comment:
We need to keep the method for compatibility with Persistence 2. We can drop it as soon as we move to Persistence >= 3.